routes/misc_map.js

```diff
@@ -53,6 +47,61 @@ module.exports = function (req, res, api, params, next) {
     t.on('error', next)
     r.pipe(t).pipe(res)
   })
+
+  function filterForkElements (elements) {
+    var latestFirst = elements.sort(cmpFork)
+    var nonForkedElements = []
+    var keepNodeRefs = {}
+    var excludeNodeRefs = {}
+    var seen = {}
+
+    // Filter out the non-latest version of each element.
+    nonForkedElements = latestFirst.filter(function (element) {
+      if (!seen[element.id]) {
+        seen[element.id] = true
+
+        // Note that all of the nodes referenced by this way should be kept.
+        if (element.type === 'way') {
+          element.nodes.forEach(function (ref) {
+            keepNodeRefs[ref] = true
+          })
+        }
+
+        return true
+      } else {
+        seen[element.id] = true
+
+        // Note that all of the nodes referenced by this way should be culled.
+        if (element.type === 'way') {
+          element.nodes.forEach(function (ref) {
+            excludeNodeRefs[ref] = true
+          })
+        }
+
+        return false
+      }
+    })
+
+    // Remove excluded entries that appear in the keep entries.
+    Object.keys(keepNodeRefs).forEach(function (ref) {
+      if (excludeNodeRefs[ref]) {
+        delete excludeNodeRefs[ref]
```

Comment:
This will be slow and can cause memory issues. I _think_ that looking at line #94 you don't actually need this, but my head is a little slow right now...